x-pack/plugins/cross_cluster_replication/public/app/components/auto_follow_pattern_form.js

```diff
@@ -49,6 +50,37 @@ const getEmptyAutoFollowPattern = (remoteClusters) => ({
   followIndexPatternSuffix: '',
 });
 
+export const updateFormValues = (fields) => ({ autoFollowPattern }) => {
```

Comment:
If you don't mind, I'd like to refactor these functions slightly to uncouple them from `setState`. In their current form, you need to understand their use case within `setState` to understand why they return functions, and I think if we uncouple them from that use case they'll be easy to understand.